presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -162,6 +162,12 @@ public PlanFragmenter(Metadata metadata, NodePartitioningManager nodePartitionin
 
     public SubPlan createSubPlans(Session session, Plan plan, boolean forceSingleNode, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector)
     {
+        return createSubPlansAndVariableAllocator(session, plan, forceSingleNode, idAllocator, warningCollector).getPlan();
+    }
+
+    public SubPlanAndVariableAllocator createSubPlansAndVariableAllocator(Session session, Plan plan, boolean forceSingleNode, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector)
```

Comment:
This code looks a little bit weird. How about instead of returning `SubPlanAndVariableAllocator` you pass the `PlanVariableAllocator` (similar to the `PlanNodeIdAllocator`)